dialer.go

```diff
@@ -151,7 +151,9 @@ func (d *customDialer) writeWARCFromConnection(reqPipe, respPipe *io.PipeReader,
 	}
 
 	if len(batch.Records) != 2 {
-		return errors.New("warc: there was a problem creating one of the WARC records")
+		err = errors.New("warc: there was an unspecified problem creating one of the WARC records")
+		d.client.errorChannel <- err
```

Comment:
And here, not returning `err` could save an entire variable, by sending `errors.New` directly through the channel.